src/Dali/RedSharp.Dali.ViewModel/ImageItem.cs

```diff
@@ -0,0 +1,195 @@
+﻿using ReactiveUI;
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.Processing;
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Text;
+
+namespace RedSharp.Dali.ViewModel
+{
+    /// <summary>
+    /// View Model for storing image.
+    /// Wrapps <see cref="Image{TPixel}"/>.
+    /// </summary>
+    public class ImageItem : ReactiveObject, IDisposable
+    {
+        #region Static
+        /// <summary>
+        /// Parameter that identifies files bufferization. I think there will 
+        /// be opportunity to set from app settings in the future.
+        /// </summary>
+        public static bool CacheImages { get; set; }
+        #endregion
+
+        #region Fields
+        private bool _isSelectd;
+        private string _path;
+        private byte[] _cache;
+        private Image _preview;
+        #endregion
+
+        #region Construction
+
+        /// <summary>
+        /// Constructs <see cref="ImageItem"/> object. Caches imput files if <see cref="ImageItem.CacheImages"/> 
+        /// property is set.
+        /// </summary>
+        /// <param name="path">Path to file with image. File should exist.</param>
+        public ImageItem(string path)
+        {
+            //We can create IOService in future to ease unit testing.
+            if (!File.Exists(path))
+                throw new ArgumentException($"{nameof(path)} is not exists.");
```

Comment:
It would be nice to add <exception> block to the comment

Reply:
What block do you mean?

Reply:
Part of comment was missed (github is the best social network), I meant an exception block with describing exception type and cases when it will threw.

Reply:
I'll add it.

Reply:
I am appreciated.